### ISSUE TITLE
Delete old blocks in archive node is probably broken

### ISSUE THREAD

**Issue by psteckler** [MinaProtocol/mina#8546]
The code to delete blocks beneath a certain block height in the archive node is probably broken. The location is `Archive_lib.Processor.delete_older_than`.

The block is deleted, but there doesn't appear to be code to NULL the `parent_id` of other blocks that refer to it. That will cause foreign key constraint violations.

The check to remove orphaned public keys doesn't mention `block_winnner_id` in blocks or the `public_key_id` in `timing_info`.

An SQL dump of mainnet, so far, is less than 5 Mb per week, so the storage requirements aren't great. (Not sure how much actual storage is needed in the database, with indexing structures and such, but probably comparable.) As I recall, the motivation for this feature was to reduce the storage burden of the archive node. Maybe the feature should be deleted, if it's not worth fixing.

**Comment by aneesharaines** [comment]
TODOL  Short term:  Remove delete

Is it better to remove the foreign key constraints?

**Comment by psteckler** [comment]
> Is it better to remove the foreign key constraints

No, those assure the integrity of the database.

**Comment by dkijania** [comment]
Yes, that's true looks like if we want to remove blocks we need to lift db constraints temporarly or find out different way, for example alter block which is supposed to be first one and convert it to genesis (from db perspective), then we can remove previous ones

**Comment by dkijania** [comment]
I would vote to delete this feature and then revisit it once storage would be a real issue
